ubuntu/Dockerfile

```diff
@@ -3,20 +3,17 @@ FROM ubuntu:20.04
 ENV LANG=C.UTF-8
 ARG VERSION=6.20.04
 
-WORKDIR /root
+WORKDIR /tmp
 
-COPY packages packages
+COPY packages packages-dev build.sh /tmp/
```

Comment:
- I see what you mean, but it does not create another layer - it erases and recreates the current layer. This is to reduce the size of the image.

- Splitting the COPY's will not have any effect on anyone building this image from scratch - which is most people including the automated build (which always starts from scratch). I don't see a benefit apart from debugging locally. For that reason alone, you can spit it manually until you arrive at a new version and then merge it again before commit.

- If people really want to find the `packages` file and understand what it's used for, they'll have to come here and look at the source of the Dockerfile, at which point they will know even more. In any case, that's an easy fix, e.g.
```patch
diff --git a/ubuntu/Dockerfile b/ubuntu/Dockerfile
index 0d845ba..4ca2d57 100644
--- a/ubuntu/Dockerfile
+++ b/ubuntu/Dockerfile
@@ -13,7 +13,7 @@ RUN apt-get update -qq \
  && ./build.sh \
  && apt-get --purge autoremove -y $(cat packages-dev) \
  && dpkg -l $(cat packages) \
- && rm -rf /var/lib/apt/lists/* /tmp/*
+ && cp packages /root/ && rm -rf /var/lib/apt/lists/* /tmp/*

 ENV PYTHONPATH=/usr/local/lib/root

```

Reply:
> I see what you mean, but it does not create another layer - it erases and recreates the current layer. This is to reduce the size of the image.

I'd like to avoid having to recreate the layers. There should be one layer for all dependencies, then another layer for ROOT, in a way that the first layer can be reused if the base image or the list of packages doesn't change.

>  I don't see a benefit apart from debugging locally. For that reason alone, you can spit it manually until you arrive at a new version and then merge it again before commit.

This is an unecessary hassle, we can have the best of both by installing only the runtime dependencies in the first layer, which can be reused later without having to change the Dockerfile. I doubt that splitting the COPY commands will increase image size.

As for the `packages` file, it's OK if we don't leave it in the image until the end when we document what it's for.